### PR TITLE
Analyze Operators + Plan Nodes

### PR DESCRIPTION
**Pull request by rohanaggarwal7997**
This is the continuation of #704 which takes the Analyze operators through the entire pipeline. From parser , binder , optimizer and planner. The final step would be to add code generation and then analyze functionality would be good to go. Two tests are provided.

This is for #716.

### CODE REVIEW

**Comment by 17zhangw** [comment]
Was any conclusion ever reached about @malin1993ml's comment on #716?

**Comment by rohanaggarwal7997** [comment]
I thought it will be unnecessary code duplication at 3 stages of making 3 different types of nodes under the analyse. It would be convinient calling three functions directly under codegen and return the result.

**Comment by malin1993ml** [comment]
I was thinking of only adding those nodes at the last stage of query
optimization to reuse the aggregation logic during codegen. But I’m also
fine if it’s already convenient to directly call the functions. I don’t
think you’ve touched the codegen part yet, so you can revisit this when you
do that.

File: src/include/optimizer/physical_operators.h [hunk 1]
```diff
@@ -2051,5 +2051,71 @@ class DropView : public OperatorNode<DropView> {
...
+   */
+  const catalog::namespace_oid_t &GetNamespaceOid() const { return namespace_oid_; }
+
+  /**
+   * @return OID of the table
+   */
+  const catalog::table_oid_t &GetTableOid() const { return table_oid_; }
+
+  /**
+   * @return trigger columns
```
Comment (apavlo) [inline src/include/optimizer/physical_operators.h]: This is not a trigger.

Comment (rohanaggarwal7997) [reply src/include/optimizer/physical_operators.h]: ya sorry for mistake, i copy pasted lot of documentation

File: src/include/optimizer/logical_operators.h [hunk 2]
```diff
@@ -1951,4 +1951,70 @@ class LogicalDropView : public OperatorNode<LogicalDropView> {
...
+   */
+  const catalog::namespace_oid_t &GetNamespaceOid() const { return namespace_oid_; }
+
+  /**
+   * @return OID of the table
+   */
+  const catalog::table_oid_t &GetTableOid() const { return table_oid_; }
+
+  /**
+   * @return trigger columns
```
Comment (17zhangw) [inline src/include/optimizer/logical_operators.h]: Should probably not be "trigger" but just "columns".

File: src/optimizer/logical_operators.cpp [hunk 3]
```diff
@@ -1192,6 +1192,40 @@ bool LogicalDropView::operator==(const BaseOperatorNode &r) {
   return node.namespace_oid_ == namespace_oid_;
 }
 
+//===--------------------------------------------------------------------===//
+// LogicalCreateTrigger
```
Comment (17zhangw) [inline src/optimizer/logical_operators.cpp]: Change from "LogicalCreateTrigger" to "LogicalAnalyze".

File: src/optimizer/physical_operators.cpp [hunk 4]
```diff
@@ -1220,6 +1220,40 @@ bool DropView::operator==(const BaseOperatorNode &r) {
   return node.namespace_oid_ == namespace_oid_;
 }
 
+//===--------------------------------------------------------------------===//
+// LogicalCreateTrigger
```
Comment (17zhangw) [inline src/optimizer/physical_operators.cpp]: Change from "LogicalCreateTrigger" to "Analyze"

File: src/optimizer/query_to_operator_transformer.cpp [hunk 5]
```diff
@@ -584,9 +584,19 @@ void QueryToOperatorTransformer::Visit(parser::CopyStatement *op, parser::ParseR
   }
 }
 
-void QueryToOperatorTransformer::Visit(UNUSED_ATTRIBUTE parser::AnalyzeStatement *op,
+void QueryToOperatorTransformer::Visit(parser::AnalyzeStatement *op,
                                        UNUSED_ATTRIBUTE parser::ParseResult *parse_result) {
   OPTIMIZER_LOG_DEBUG("Transforming AnalyzeStatement to operators ...");
+  std::vector<catalog::col_oid_t> columns;
+  auto tb_oid = accessor_->GetTableOid(op->GetAnalyzeTable()->GetTableName());
```
Comment (17zhangw) [inline src/optimizer/query_to_operator_transformer.cpp]: If there is a namespace specified, shouldn't this instead use:
`CatalogAccessor::GetTableOid(namespace_oid_t ns, std::string name)`

Comment (rohanaggarwal7997) [reply src/optimizer/query_to_operator_transformer.cpp]: A similar thing is done in create trigger. Does it make a difference?

**Comment by rohanaggarwal7997** [comment]
Addressed all comments

**Comment by apavlo** [comment]
@rohanaggarwal7997 If you have addressed all the comments, mark them as 'resolved' on Github.

**Comment by rohanaggarwal7997** [comment]
> @rohanaggarwal7997 If you have addressed all the comments, mark them as 'resolved' on Github.

done

File: src/include/optimizer/logical_operators.h [hunk 6]
```diff
@@ -1951,4 +1951,70 @@ class LogicalDropView : public OperatorNode<LogicalDropView> {
...
+ private:
+  /**
+   * OID of the database
+   */
+  catalog::db_oid_t database_oid_;
+
+  /**
+   * OID of the namespace
+   */
+  catalog::namespace_oid_t namespace_oid_;
```
Comment (mbutrovich) [inline src/include/optimizer/logical_operators.h]: Tables should be unique within a database so I don't think we need namespace oid here.

File: src/include/optimizer/physical_operators.h [hunk 7]
```diff
@@ -2051,5 +2051,71 @@ class DropView : public OperatorNode<DropView> {
...
+ private:
+  /**
+   * OID of the database
+   */
+  catalog::db_oid_t database_oid_;
+
+  /**
+   * OID of the namespace
+   */
+  catalog::namespace_oid_t namespace_oid_;
```
Comment (mbutrovich) [inline src/include/optimizer/physical_operators.h]: See previous namespace oid comment.

File: src/optimizer/query_to_operator_transformer.cpp (hunk 5 above)
Comment (rohanaggarwal7997) [reply src/optimizer/query_to_operator_transformer.cpp]: Now there are no more namespace id

File: src/include/optimizer/physical_operators.h (hunk 7 above)
Comment (rohanaggarwal7997) [reply src/include/optimizer/physical_operators.h]: Addressed

File: src/include/optimizer/logical_operators.h (hunk 6 above)
Comment (rohanaggarwal7997) [reply src/include/optimizer/logical_operators.h]: Addressed

**Comment by rohanaggarwal7997** [comment]
I dont know why but there were lot of changes in the optimizer for some reason of renamings. I had to rename almost all lines that I added. It works now. I also removed all namespace related things. Both the tests still pass. I think its good to merge.

**Comment by apavlo** [comment]
@rohanaggarwal7997 Fix the linter error.

**Comment by rohanaggarwal7997** [comment]
> @rohanaggarwal7997 Fix the linter error.

Its actually not a linter error. I did not knew that we have to check warnings.txt after running doxygen.

**Pull request merged**